torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -2769,6 +2770,52 @@ def test_DistributedDataParallel_non_default_stream(self):
                         msg=f"Expected gradient of {expected_grad} but got {avg} on rank {self.rank}",
                     )
 
+        @unittest.skipIf(
+            BACKEND != "nccl" and BACKEND != "gloo",
+            "Only NCCL and GLOO backend support DistributedDataParallel",
+        )
+        @skip_if_lt_x_gpu(int(os.environ["WORLD_SIZE"]))
+        @skip_if_rocm
+        def test_DistributedDataParallel_powerSGD_ddp_comm_hook(self):
+            stream = torch.cuda.Stream(self.rank)
+            rank = self.rank
+            with torch.cuda.stream(stream):
+                net = torch.nn.parallel.DistributedDataParallel(
+                    torch.nn.Linear(1, 5).to(rank), device_ids=[rank]
+                )
+                process_group = torch.distributed.new_group([0, 1])
```

Comment:
Should we test with the process_group being the entire world as well?

Reply:
I just followed some examples like `test_ddp_uneven_inputs_replicated_error` in the same file, and I didn't see it's necessary to testing both cases here.

I can change `torch.distributed.new_group([0, 1])` to `list(range(0, dist.get_world_size()))`, or just use `group, _, rank = self._init_global_test()`.

I plan to rewrite the unit test in a separate PR, as mentioned in some other comments.